File: ts/src/nougen_shards/brain_scan/registry.ts
/**
 * Static term sets and root path registry. (TS mimic of brain_scan/registry.py)
 *
 * Python `set` literals -> Set<string>; Python `list` literals -> string[].
 * pathlib Paths under Path.home() -> absolute path strings via path.join(homedir(), ...).
 */
import { homedir } from "node:os";
import * as path from "node:path";

// Known source adapters / tools
export const KNOWN_TOOLS: string[] = [
  "gemini", "claude", "codex", "cursor", "continue", "copilot",
  "openhands", "mem0", "ollama", "qwen", "roo", "vscode", "github",
];

export const GLOBAL_ROOTS: string[] = [
  homedir(),
  path.join(homedir(), ".claude"),
  path.join(homedir(), ".codex"),
  path.join(homedir(), ".gemini"),
  path.join(homedir(), ".cursor"),
  path.join(homedir(), ".continue"),
  path.join(homedir(), ".copilot"),
  path.join(homedir(), ".openhands"),
  path.join(homedir(), ".mem0"),
  path.join(homedir(), ".ollama"),
  path.join(homedir(), ".qwen"),
  path.join(homedir(), ".roo"),
  path.join(homedir(), ".vscode"),
  path.join(homedir(), ".vscode-shared"),
];

export const PROJECT_ROOT_NAMES: string[] = [
  ".agent", ".agents", ".claude", ".codex", ".gemini", ".cursor",
  ".continue", ".openhands", ".roo", ".vscode", ".github", ".logs",
];

export const PROJECT_FILES: string[] = [
  "AGENTS.md", "CLAUDE.md", "GEMINI.md", "README.md",
];

export const DANGER_ZONES: Set<string> = new Set([
  ".ssh", ".aws", ".azure", ".config", ".gnupg", ".kube",
  "credentials", "1password", "bitwarden",
]);

export const SKIP_DIRS: Set<string> = new Set([
  "node_modules", ".git", ".venv", "venv", "__pycache__", ".pytest_cache",
  ".cache", ".npm", ".pnpm-store", ".pnpm-cache", ".bun", ".m2", ".nuget",
  ".gradle", ".docker", "dist", "build", "target", "out",
]);

export const SUPPORTED_EXTS: Set<string> = new Set([
  ".json", ".jsonl", ".zst", ".md", ".txt", ".log", ".toml",
  ".yaml", ".yml", ".sqlite", ".db",
]);

export const HIGH_SIGNAL_TERMS: Set<string> = new Set([
  "conversation", "session", "transcript", "chat", "messages",
  "memory", "rules", "agents", "instructions", "checkpoint",
  "rollout", "tool_calls", "tasks", "history",
]);

export const MEDIUM_SIGNAL_TERMS: Set<string> = new Set([
  "settings", "config", "workspace", "launch", "commands",
  "logs", "debug", "workflow",
]);

export const LOW_SIGNAL_TERMS: Set<string> = new Set([
  "cache", "lockfile", "package", "compiled", "binary",
  "image", "video", "audio", "weights",
]);

File: src/nougen_shards/brain_scan/registry.py
from pathlib import Path

# Known source adapters / tools
KNOWN_TOOLS = [
    "gemini", "claude", "codex", "cursor", "continue", "copilot",
    "openhands", "mem0", "ollama", "qwen", "roo", "vscode", "github"
]

GLOBAL_ROOTS = [
    Path.home(),
    Path.home() / ".claude",
    Path.home() / ".codex",
    Path.home() / ".gemini",
    Path.home() / ".cursor",
    Path.home() / ".continue",
    Path.home() / ".copilot",
    Path.home() / ".openhands",
    Path.home() / ".mem0",
    Path.home() / ".ollama",
    Path.home() / ".qwen",
    Path.home() / ".roo",
    Path.home() / ".vscode",
    Path.home() / ".vscode-shared"
]

# Agent memory stores that do not live under a dot-directory named after their
# tool, so the original GLOBAL_ROOTS never reached them. Appended to
# GLOBAL_ROOTS rather than kept as a separate list so that anything patching
# GLOBAL_ROOTS (the test suite pins it to a temp dir) still controls every root
# that gets walked — a second, unpatched root list would silently escape that
# isolation and reach the real home directory.
EXTRA_DB_ROOTS = [
    Path.home() / "Watchtower",
    Path.home() / ".iris",
]

GLOBAL_ROOTS.extend(EXTRA_DB_ROOTS)

# SQLite is read via query, not by slurping the file, so these extensions are
# exempt from the byte-size ceiling that applies to text candidates.
SQLITE_EXTS = {".db", ".sqlite", ".sqlite3"}

# NouGenShards' own substrate. Ingesting it would re-import every shard as the
# body of a new shard on each run.
SELF_EXCLUDE_DIRS = {".nougen"}

PROJECT_ROOT_NAMES = [
    ".agent", ".agents", ".claude", ".codex", ".gemini", ".cursor",
    ".continue", ".openhands", ".roo", ".vscode", ".github", ".logs"
]

PROJECT_FILES = [
    "AGENTS.md", "CLAUDE.md", "GEMINI.md", "README.md"
]

DANGER_ZONES = {
    ".ssh", ".aws", ".azure", ".config", ".gnupg", ".kube", 
    "credentials", "1password", "bitwarden"
}

SKIP_DIRS = {
    "node_modules", ".git", ".venv", "venv", "__pycache__", ".pytest_cache",
    ".cache", ".npm", ".pnpm-store", ".pnpm-cache", ".bun", ".m2", ".nuget",
    ".gradle", ".docker", "dist", "build", "target", "out",
    "antigravity", "antigravity-backup", "antigravity-ide", "antigravity-browser-profile",
    "extensions", "cacheddata", "cachedextensions", "bin", "locales", "packages", "usr", "lib"
}

SUPPORTED_EXTS = {
    ".json", ".jsonl", ".md", ".txt", ".log", ".toml", 
    ".yaml", ".yml"
}

HIGH_SIGNAL_TERMS = {
    "conversation", "session", "transcript", "chat", "messages", 
    "memory", "rules", "agents", "instructions", "checkpoint", 
    "rollout", "tool_calls", "tasks", "history"
}

MEDIUM_SIGNAL_TERMS = {
    "settings", "config", "workspace", "launch", "commands", 
    "logs", "debug", "workflow"
}

LOW_SIGNAL_TERMS = {
    "cache", "lockfile", "package", "compiled", "binary", 
    "image", "video", "audio", "weights"
}
